jcl/src/java.base/share/classes/java/lang/StringBuilder.java

```diff
@@ -209,6 +209,7 @@ public StringBuilder append (char[] chars) {
 			// Check if the StringBuilder is compressed
 			if (count >= 0) {
 				decompress(newLength);
+				currentCapacity = capacityInternal();
```

Comment:
Could we instead modify the `decompress` method to return the new capacity? We can then avoid these additional calls to `capacityInternal()` since the `decompress` method already knows what the new capacity will be.